services/profiling.py

```diff
@@ -79,7 +79,12 @@ def _get_critical_files_from_yaml(
             "critical_files_paths"
         ):
             return []
-        commit_sha = self.commit_sha or profiling_commit.commit_sha
+
+        if profiling_commit is not None:
```

Comment:
Was there something wrong with the one on the left? Or is this mostly preference?

Reply:
Hmm good point this change is actually not equivalent, old code gives `self.commit_sha` preference whereas as this gives `profiling_commit.commit_sha` preference. I need to do it like this:
```
        commit_sha = None
        if self.commit_sha:
            commit_sha = self.commit_sha
        elif profiling_commit:
            commit_sha = profiling_commit.commit_sha
```
mypy didn't like the old style because `profiling_commit` might be null so it can produce an error